common/djangoapps/enrollment/api.py

```diff
@@ -225,7 +225,8 @@ def update_enrollment(user_id, course_id, mode=None, is_active=None):
         }
 
     """
-    _validate_course_mode(course_id, mode)
+    if mode is not None:
+        _validate_course_mode(course_id, mode)
```

Comment:
OK, got it.  I didn't see that `mode` gets defaulted in the HTTP request.  The behavior is still different for the api function itself however, `mode` effectively defaults to "honor" as a result of this change.  I'll settle for a docstring update :-)